### PR TITLE
Implement daemon

### PR DESCRIPTION
**Pull request by Naatan**

### CODE REVIEW

**Review by daved — DISMISSED** [review]
I do think most of my comments are worth the time to consider, but nothing tipped the scale. Great work!

File: cmd/state-svc/internal/server/server.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/99designs/gqlgen/graphql/handler"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+	"github.com/99designs/gqlgen/graphql/handler/lru"
+	"github.com/99designs/gqlgen/graphql/handler/transport"
+	"github.com/labstack/echo/v4"
```
[the remark this answers is not in the document]
Comment (Naatan) [reply cmd/state-svc/internal/server/server.go:12]: https://www.pivotaltracker.com/story/show/177418190

File: cmd/state-svc/internal/server/server.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	s.graphServer = newGraphServer()
+	s.listener = listener
+	s.httpServer = newHTTPServer(listener)
+
+	s.setupRouting()
+
+	return s, nil
+}
+
+func (s *Server) Port() (int, error) {
```
[the remark this answers is not in the document]
Comment (Naatan) [reply cmd/state-svc/internal/server/server.go:42]: Neat! Didn't know about this.

File: cmd/state-svc/main.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	if err != nil {
+		return errs.Wrap(err, "Could not create server")
+	}
+
+	port, err := s.Port()
+	if err != nil {
+		return errs.Wrap(err, "Could not detect port")
+	}
+
+	cfg, err := config.New()
```
[the remark this answers is not in the document]
Comment (Naatan) [reply cmd/state-svc/main.go:33]: You'd have to call `cfg.Reload()` explicitly.

File: cmd/state-svc/main.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+	go func() {
+		oscall := <-c
+		logging.Debug("system call:%+v", oscall)
+		if err := s.Close(); err != nil {
+			fmt.Fprintf(os.Stderr, "Closing server failed: %w", err)
+		}
+	}()
+
+	if err := s.Start(); err != nil {
```
[the remark this answers is not in the document]
Comment (Naatan) [reply cmd/state-svc/main.go:53]: This is effectively just aliasing `echo.Start()`. Not much else we can do.

File: go.mod [hunk 5]
```diff
@@ -78,14 +81,15 @@ require (
...
 	github.com/vbauerster/mpb/v4 v4.10.0
+	github.com/vektah/gqlparser/v2 v2.1.0
 	github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8 // indirect
-	golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
+	golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
 	golang.org/x/net v0.0.0-20201021035429-f5854403a974
 	golang.org/x/sys v0.0.0-20201201145000-ef89a241ccb3
 	golang.org/x/text v0.3.3
 	google.golang.org/genproto v0.0.0-20200815001618-f69a88009b70
+	google.golang.org/grpc v1.36.0 // indirect
```
[the remark this answers is not in the document]
Comment (Naatan) [reply go.mod:91]: The google SDK's we use for CI integration still use grpc. I did run `go mod tidy`, but just tried it again just to be safe and there are no further changes.

File: pkg/platform/api/svc/request/request.go [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package request
+
+type Request interface {
+	Query() string
+	Vars() map[string]interface{}
+}
+
+type Client interface {
+	Run(request Request, response interface{}) error
```
[the remark this answers is not in the document]
Comment (Naatan) [reply pkg/platform/api/svc/request/request.go]: This was left over from some experimenting I was doing, I can remove it. Thanks for catching!

File: cmd/state-svc/main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (daved) [reply cmd/state-svc/main.go:49]: I meant that the error message might be more clear if it states something like "Error encountered while serving".

**Review by daved — APPROVED** [review]

**Pull request merged**